src/corelibs/U2Algorithm/src/registry/TempCalcRegistry.h

```diff
@@ -0,0 +1,76 @@
+/**
+ * UGENE - Integrated Bioinformatics Tools.
+ * Copyright (C) 2008-2022 UniPro <ugene@unipro.ru>
+ * http://ugene.net
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+
+#ifndef _U2_TEMP_CALC_REGISTRY_
+#define _U2_TEMP_CALC_REGISTRY_
+
+#include <U2Core/IdRegistry.h>
+#include <U2Core/global.h>
+
+#include <U2Algorithm/BaseTempCalc.h>
+#include <U2Algorithm/TempCalcFactory.h>
+
+namespace U2 {
+
+/**
+ * @TempCalcRegistry is a class to registering for all methods of temperature calculation
+ */
+class U2ALGORITHM_EXPORT TempCalcRegistry : public IdRegistry<TempCalcFactory> {
+public:
+    bool registerEntry(TempCalcFactory* t) override;
+    /**
+     * Get the default temperature calculator
+     * Get the factory with the corresponding ID @saveId
+     * If @saveId is empty - get the first registred factory
+     * @return pointer to the temperature calculator
+     */
+    BaseTempCalc* getDefaultTempCalculator(const QString& saveId = "") const;
```

Comment:
It is a good example where you should return a shared pointer instead of raw pointer. I see that the code below has a lot of non-related changes just because it has to deal with memory management.

Reply:
OK